src/poisson.f90

```diff
@@ -298,89 +298,123 @@ subroutine poisson_000(rhs)
 
     integer :: nx,ny,nz, i,j,k
 
-    complex(mytype) :: cx
-    real(mytype) :: rl, iy
-    external cx, rl, iy
+    complex(mytype) :: cw1_tmp
+    real(mytype) :: ntot
+#ifdef DEBUG
+    integer, dimension(3) :: dim3d
+#endif
 
     nx = nx_global
     ny = ny_global
     nz = nz_global
+    ntot = real(nx, kind=mytype) &
+         * real(ny, kind=mytype) &
+         * real(nz, kind=mytype)
 
     if (.not. fft_initialised) then
        call decomp_2d_fft_init(PHYSICAL_IN_Z)
        fft_initialised = .true.
     end if
 
     ! compute r2c transform
+#ifdef DEBUG
```

Comment:
Could you add a verbose flag set to false by default to avoid printing plenty of numbers when running the code compiled with the DEBUG preprocessor directive ?

Reply:
In theory we pring only 8 values since I am using a large index jump. Adding an extra flag will be a bit too much since under DEBUG there is almost no code.